rest-tests/src/test/java/org/hawkular/metrics/test/InfluxDatabaseITest.java

```diff
@@ -0,0 +1,231 @@
+                                            TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(series);
+        Assert.assertEquals(serieSource.toString(), series.get(0).toString());
+    }
+
+    @Test
+    public void testQueryStddev(){
+        String timeSeries = "queryTestStddev";
+        long startTime  = System.currentTimeMillis();
+        Serie generatedSerie = getSerie(timeSeries, startTime);
+        influxDB.write(dbName, TimeUnit.MILLISECONDS, generatedSerie);
+        startTime = startTime-9000L;
+        Serie serieSource = new Serie.Builder(timeSeries)
+        .columns("time", "stddev")
+        .values(new Double(startTime+8000L), 2.7386127875258297)
+        .build();
+        List<Serie> series = influxDB.query(dbName,
+                                            "select stddev(value) from "+timeSeries
+                                            +" where time > now() - 30s group by time(30s)",
+                                            TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(series);
+        Assert.assertEquals(serieSource.toString(), series.get(0).toString());
+    }
+
+    @Test
+    public void testQueryMean(){
+        String timeSeries = "queryTestMean";
+        long startTime  = System.currentTimeMillis();
+        Serie generatedSerie = getSerie(timeSeries, startTime);
+        influxDB.write(dbName, TimeUnit.MILLISECONDS, generatedSerie);
+        startTime = startTime-9000L;
+        Serie serieSource = new Serie.Builder(timeSeries)
+        .columns("time", "mean")
+        .values(new Double(startTime+8000L), 14.9D)
+        .build();
+        List<Serie> series = influxDB.query(dbName,
+                                            "select mean(value) from "+timeSeries
+                                            +" where time > now() - 30s group by time(30s)",
+                                            TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(series);
+        Assert.assertEquals(serieSource.toString(), series.get(0).toString());
+    }
+
+    @Test
+    public void testQuerySum(){
+        String timeSeries = "queryTestSum";
+        long startTime  = System.currentTimeMillis();
+        Serie generatedSerie = getSerie(timeSeries, startTime);
+        influxDB.write(dbName, TimeUnit.MILLISECONDS, generatedSerie);
+        startTime = startTime-9000L;
+        Serie serieSource = new Serie.Builder(timeSeries)
+        .columns("time", "sum")
+        .values(new Double(startTime+8000L), 134.1D)
+        .build();
+        List<Serie> series = influxDB.query(dbName,
+                                            "select sum(value) from "+timeSeries
+                                            +" where time > now() - 30s group by time(30s)",
+                                            TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(series);
+        Assert.assertEquals(serieSource.toString(), series.get(0).toString());
```

Comment:
Ok now I see the issue. You can't compare the two series as string. You may create an `#assertSeriesEquals` method, in this class or its parent. In this method you could iterate over the points and check that the timestamps and values are "equal" within a delta.

See http://junit.org/javadoc/latest/org/junit/Assert.html#assertEquals%28java.lang.String,%20double,%20double,%20double%29